elephant/spike_train_generation.py

```diff
@@ -461,13 +482,50 @@ def inhomogeneous_poisson_process(rate, as_array=False):
         signal=rate, times=homogeneous_poiss.magnitude *
         homogeneous_poiss.units)
     # Accept each spike at time t with probability rate(t)/max_rate
-    u = np.random.uniform(size=len(homogeneous_poiss)) * max_rate
-    spikes = homogeneous_poiss[u < rate_interpolated.flatten()]
-    if as_array:
-        spikes = spikes.magnitude
+    random_uniforms = np.random.uniform(size=len(homogeneous_poiss)) * max_rate
+    spikes = homogeneous_poiss[random_uniforms < rate_interpolated.flatten()]
+
+    if refractory_period is not None:
+        refractory_period = refractory_period.rescale(
+            rate.t_stop.units).magnitude
+        spikes = _thinning_for_refractory_period(spikes.magnitude,
```

Comment:
doing this reduces the expected firing rate
are you sure this is the right way to include a refractory period in inhomogeneous PP?

Reply:
It is maybe a bit confusing, but I first updated the firing rate to an effective firing rate. In line 471. And then I thin it out. In average both effects cancel to result in the wished average firing rate.

Reply:
In test_statistics.py the spike-count is checked against its expectation. It works well.